src/main/java/net/snowflake/ingest/streaming/internal/ParquetValueParser.java

```diff
@@ -91,25 +95,27 @@ static ParquetBufferValue parseColumnValueToParquet(
                   defaultTimezone);
           value = longValue;
           stats.addIntValue(BigInteger.valueOf(longValue));
-          size = 8;
+          estimatedParquetSize += 8;
           break;
         case DOUBLE:
           double doubleValue =
               DataValidationUtil.validateAndParseReal(columnMetadata.getName(), value);
           value = doubleValue;
           stats.addRealValue(doubleValue);
-          size = 8;
+          estimatedParquetSize += 8;
           break;
         case BINARY:
           if (logicalType == AbstractRowBuffer.ColumnLogicalType.BINARY) {
             value = getBinaryValueForLogicalBinary(value, stats, columnMetadata);
-            size = ((byte[]) value).length;
+            estimatedParquetSize += ((byte[]) value).length;
           } else {
             String str = getBinaryValue(value, stats, columnMetadata);
             value = str;
             if (str != null) {
-              size = str.getBytes().length;
+              estimatedParquetSize += str.getBytes().length;
             }
+            estimatedParquetSize +=
+                4; // Parquet stores length in 4 bytes before the actual data bytes
           }
```

Comment:
```
        int length = 0;
        if (logicalType == AbstractRowBuffer.ColumnLogicalType.BINARY) {
          value = getBinaryValueForLogicalBinary(value, stats, columnMetadata);
          length = ((byte[]) value).length;
        } else {
          String str = getBinaryValue(value, stats, columnMetadata);
          value = str;
          if (str != null) {
            length = str.getBytes().length;
          }
        }
        if (value != null) {
          // Parquet stores length in 4 bytes before the actual data bytes
          estimatedParquetSize += 4 + length;
        }
```
Otherwise 4 is not counted for `ColumnLogicalType.BINARY` but counted for str that is null.

Reply:
done.